src/wallet/wallet2.cpp

```
@@ -3901,24 +4074,43 @@ bool wallet2::store_keys(const fs::path& keys_file_name, const epee::wipeable_st
+    unlock_keys_file();
+  else
+    unlock_background_keys_file();
+
   std::error_code e;
 #ifdef WIN32
   // std::filesystem::rename is broken on Windows and fails if the file already exists
   fs::remove(keys_file_name, e);
 #endif
   fs::rename(tmp_file_name, keys_file_name, e);
-  lock_keys_file();
+  if (background_keys_file)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)